vscode/esbuild.js

```diff
@@ -28,7 +30,21 @@ const watchConfig = {
       },
     },
   };
-  
+
+const NON_NPM_ARTIFACTORY = new RegExp(
+  String.raw`"resolved": "(?!https://registry.npmjs.org).*"`,
```

Comment:
Minor regex suggestions for being whitespaces and quotes agnostic.
```re
("|')resolved\1\s*:\s*("|')http[s]*://(?!registry.npmjs.org)[^"']+\2
```
Do test the suggested regex for functional correctness if you choose to use it.

Reply:
Since it is searching for the pattern in `package-lock.json`. It would have double quotes only, correct?
So, should I change `("|')` to `(")`

Reply:
Yes, you're right. The package-lock.json is expected to be from a code generator, not created manually. So, it is expected to stick to the JSON standard of double-quotes. 
In general, however, when reading any input data, it is necessary to be more lenient. In that case, it would have been more appropriate to use a JSON parser and test the values.
So please go ahead with the simpler regex like:
```re
"resolved"\s*:\s*"http[s]*://(?!registry.npmjs.org)[^"]+"
```
Thanks.